controllers/knative/knative.go

```diff
@@ -84,3 +102,100 @@ func GetKnativeAvailability(cfg *rest.Config) (*Availability, error) {
+				return nil, fmt.Errorf("error reading the platform referred by the cluster platform")
+			}
+			if platform.Spec.Eventing != nil {
+				return platform.Spec.Eventing.Broker, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+const knativeBrokerAnnotation = "eventing.knative.dev/broker.class"
+
+func GetKnativeResource(ctx context.Context, cfg *rest.Config, kRef *duckv1.KReference) (*unstructured.Unstructured, error) {
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	gv, err := schema.ParseGroupVersion(kRef.APIVersion)
+	if err != nil {
+		return nil, err
+	}
+	resourceId := schema.GroupVersionResource{
+		Group:    gv.Group,
+		Version:  gv.Version,
+		Resource: kRef.Kind,
+	}
+	if len(kRef.Namespace) == 0 {
+		return nil, fmt.Errorf("namespace for knative resource %s is missing", kRef.Name)
+	}
+	list, err := dynamicClient.Resource(resourceId).Namespace(kRef.Namespace).List(ctx, metav1.ListOptions{})
+	fmt.Printf("list:%v", list)
+	return dynamicClient.Resource(resourceId).Namespace(kRef.Namespace).Get(ctx, kRef.Name, metav1.GetOptions{})
+}
+
+func IsKnativeBroker(kRef *duckv1.KReference) bool {
+	return kRef.APIVersion == "eventing.knative.dev/v1" && kRef.Kind == "Broker"
+}
+
+func IsKnativeEnvInjected(ctx context.Context, c client.Client, deploymentName, namespace string) (bool, error) {
+	deployment := &appsv1.Deployment{}
+	if err := c.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: namespace}, deployment); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil //deployment not found
+		}
+		return false, err
+	}
+	for _, env := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if env.Name == KSink {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func IsDataIndexEnabled(plf *operatorapi.SonataFlowPlatform) bool {
+	if plf.Spec.Services != nil {
+		if plf.Spec.Services.DataIndex != nil {
+			return pointer.BoolDeref(plf.Spec.Services.DataIndex.Enabled, false)
+		}
+		return false
+	}
+	// Check if DataIndex is enabled in the platform status
+	if plf.Status.ClusterPlatformRef != nil && plf.Status.ClusterPlatformRef.Services != nil && plf.Status.ClusterPlatformRef.Services.DataIndexRef != nil && len(plf.Status.ClusterPlatformRef.Services.DataIndexRef.Url) > 0 {
+		return true
+	}
+	return false
+}
+
+func IsJobServiceEnabled(plf *operatorapi.SonataFlowPlatform) bool {
+	if plf.Spec.Services != nil {
+		if plf.Spec.Services.JobService != nil {
+			return pointer.BoolDeref(plf.Spec.Services.JobService.Enabled, false)
+		}
+		return false
+	}
+	// Check if JobService is enabled in the platform status
+	if plf.Status.ClusterPlatformRef != nil && plf.Status.ClusterPlatformRef.Services != nil && plf.Status.ClusterPlatformRef.Services.JobServiceRef != nil && len(plf.Status.ClusterPlatformRef.Services.JobServiceRef.Url) > 0 {
+		return true
+	}
+	return false
```

Comment:
Can you please move the functions `IsDataIndexEnabled` and `IsJobServiceEnabled` to the `controllers/platform` module?

Reply:
Sounds good. Done.